rundeckapp/grails-spa/packages/ui-trellis/src/app/components/activity/types/activityTypes.ts

```diff
@@ -2,3 +2,84 @@ export interface DateFilterProps {
   enabled: boolean;
   datetime: string;
 }
+export type ComponentStub = boolean | { template: string };
+export type Stubs = string[] | Record<string, ComponentStub>;
+export interface Directive {
+  beforeMount?: (
+    el: HTMLElement,
+    binding: any,
+    vnode: any,
+    prevVnode: any,
+  ) => void;
+  mounted?: (el: HTMLElement, binding: any, vnode: any, prevVnode: any) => void;
+  beforeUpdate?: (
+    el: HTMLElement,
+    binding: any,
+    vnode: any,
+    prevVnode: any,
+  ) => void;
+  updated?: (el: HTMLElement, binding: any, vnode: any, prevVnode: any) => void;
+  beforeUnmount?: (
+    el: HTMLElement,
+    binding: any,
+    vnode: any,
+    prevVnode: any,
+  ) => void;
+  unmounted?: (
+    el: HTMLElement,
+    binding: any,
+    vnode: any,
+    prevVnode: any,
+  ) => void;
+}
+export interface GlobalOptions {
+  stubs?: Stubs;
+  mocks?: Record<string, unknown>;
+  directives?: Record<string, Directive>;
+}
```

Comment:
Please avoid mixing the types being used in a component with the types being used in tests. If these types are being used, please move them to their own types file inside the tests folder.